tests/unit/s2n_x509_validator_test.c

```diff
@@ -585,11 +577,10 @@ int main(int argc, char **argv)
         uint8_t *chain_data = s2n_stuffer_raw_read(&cert_chain_stuffer, chain_len);
         EXPECT_NOT_NULL(chain_data);
 
-        /* The default cert chain includes a SHA1 signature, so the security policy must allow SHA1 cert signatures. */
         EXPECT_SUCCESS(s2n_connection_set_cipher_preferences(connection, "default"));
 
         /* alter a random byte in the certificate to make it invalid */
-        chain_data[500] = (uint8_t) (chain_data[500] << 2);
+        chain_data[200] = (uint8_t) (chain_data[200] << 2);
```

Comment:
This test is trying to alter the "end-entity certificate" to make it invalid. I believe that is stored earlier in the chain_data[] array so using `chain_len - 1` did not work. If I use smaller values like `5`, it then throws `error decoding certificate` presumably because it messes up the header information and now doesn't know how to decode the cert. I'm not sure if there is a good way to skip this header bytes accurately

Reply:
Got it okay. In that case, could we at least assert that 200 doesn't index past `chain_data`?
```suggestion
        EXPECT_TRUE(chain_len > 200);
        chain_data[200] = (uint8_t) (chain_data[200] << 2);
```